src/librustc_mir/interpret/operand.rs

```diff
@@ -772,4 +772,16 @@ impl<'a, 'mir, 'tcx, M: Machine<'a, 'mir, 'tcx>> EvalContext<'a, 'mir, 'tcx, M>
         })
     }
 
+    /// This is used by [priroda](https://github.com/oli-obk/priroda) to get an OpTy from a local
+    pub fn read_local_of_frame(
+        &self,
+        frame: &super::Frame<'mir, 'tcx>,
+        local: mir::Local
+    ) -> EvalResult<'tcx, OpTy<'tcx>> {
+        let op = *frame.locals[local].access()?;
+        let local_ty = frame.mir.local_decls[local].ty;
+        let local_ty = self.monomorphize(local_ty, frame.instance.substs);
+        let layout = self.layout_of(local_ty)?;
+        Ok(OpTy { op, layout })
+    }
```

Comment:
Why don't you use `stack()`? `locals` is public, so that should give you all you need?

Reply:
That gives an `Operand` while I need an `OpTy` which cant be made outside rustc_mir.

Reply:
Oh I see.  In that case, please use `layout_of_local`, and please change `eval_place_to_operand` to use your new method.

Reply:
`layout_of_local` expects a frame id, while this takes an arbitrary Frame. I can do the other thing though.